vllm/model_executor/layers/attention.py

```diff
@@ -355,7 +355,14 @@ def forward(
 
         # Apply rotary embedding to the query and key before passing them
         # to the attention op.
-        query, key = self.rotary_emb(positions, query, key)
+        if not isinstance(self.rotary_emb,
+                          DynamicNTKScalingRotaryEmbeddingQwen):
+            query, key = self.rotary_emb(positions, query, key)
+        else:
+            # need to pass in prompt_token_length to compute dynamic_ntk for qwen
+            query, key = self.rotary_emb(
+                positions, query, key,
+                input_metadata.seq_groups[0][1].prompt_token_length)
```

Comment:
I have tested the code here. When there is a mixed prompt token length, there will be an unexpect result. For example, when the prompt_token_length of seq_groups is [800, 900, 1000, 2000, 5860], the token length obtained here is 800. Therefore, it will use the calculation of ntk_alpha and base within the range of 2048. This will result in abnormal output for seq with a token length of 5860.
<img width="514" alt="image" src="https://github.com/vllm-project/vllm/assets/912204/72dc837a-3c83-408d-ba56-873470723b06">